packages/connections/src/stores/connections-store.ts

```diff
@@ -141,6 +146,49 @@ async function loadConnections(
   }
 }
 
+function trackConnectionAttemptEvent({ favorite, lastUsed }: ConnectionInfo): void {
+  const trackEvent = {
+    is_favorite: Boolean(favorite),
+    is_recent: Boolean(lastUsed && !favorite),
+    is_new: !lastUsed,
+  };
+  track('Connection Attempt', trackEvent);
+}
+
+async function trackNewConnectionEvent(dataService: DataService, connectionString: string): Promise<void> {
+  const {
+    dataLake,
+    genuineMongoDB,
+    host,
+    build,
+  } = await dataService.instance();
+  const { hosts: [hostName] } = new ConnectionString(connectionString);
+  const { isAws, isAzure, isGcp } = await getCloudInfo(hostName).catch((err: Error) => {
+    debug('getCloudInfo failed', err);
+    return {};
+  });
+  const connectionOptions = dataService.getMongoClientConnectionOptions();
```

Comment:
with the new connection the `dataService.getMongoClientConnectionOptions` is not be necessary anymore, that same info can be derived from the connection string directly with `.searchParams.get('authMechanism')`

Also .. what about username and password, for that the authMechanism may not be set, how do we track that?